src/NSwag.CodeGeneration.CSharp/Templates/Client.Class.liquid

```diff
@@ -127,7 +127,7 @@
 
 {%         endif -%}
 {%     endfor -%}
-{%     if operation.HasContent and operation.ContentParameter.IsNullable != true -%}
+{%     if operation.HasContent and operation.ContentParameter.IsRequired -%}
```

Comment:
I'm not sure about that, mainly because "required" means that "null" is allowed - ie required says that a value is required but null is also a valid value.

Reply:
Incorrect, the spec says it's nullable int64 aka `long?`. The produced Client method has the correct signature:

`public async System.Threading.Tasks.Task TestOpAsync(long? body, System.Threading.CancellationToken cancellationToken)`

but the not null guard is added which doesn't make sense for a nullable parameter.

```
if (body == null)
    throw new System.ArgumentNullException("body");
```

If body IS required:

```
{
  "openapi": "3.0.1",
  "paths": {
    "/testOp": {
      "get": {
        "operationId": "testOp",
        "requestBody": {
          "content": {
            "*/*": {
              "schema": {
                "type": "integer",
                "format": "int64"
              }
            }
          },
          "required": true
        },
        "responses": {
          "200": {
            "description": "Success"
          }
        }
      }
    }
  }
}
```

then the generated signature is also correct

`public async System.Threading.Tasks.Task TestOpAsync(long body, System.Threading.CancellationToken cancellationToken)`

the null guard still is added which is not needed for non-nullable parameters but it doesn't break the logic. My pull request fixes the first case. Removes null guard for the optional body parameter.

Reply:
@RicoSuter bump

Reply:
Same PR: https://github.com/RicoSuter/NSwag/pull/3358
Ref: https://github.com/RicoSuter/NSwag/pull/3195/files

Reply:
Seems that some months ago it was like that...

Reply:
thanks!